### PR TITLE
Fixes #1596: Add test for LocationWiseTimeService.java

### LINKED ISSUE #1596: Add test for LocationWiseTimeService.java
- Issue type: Feature request

### Short description

The proposed request is for adding test for LocationWiseTimeService.java to increase test coverage.

### Usecase

It should test the LocationWiseTimeService.java while running ```./gradlew build```
### Description

The goal is to increase the test coverage of loklak.

### PR DESCRIPTION
**Pull request by simsausaurabh**
### Short description
I have:
- [X] There is a corresponding issue for this pull request.
- [X] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [X] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

Fixes: #1596

### CODE REVIEW

**Comment by simsausaurabh** [comment]
@singhpratyush please review.

File: test/org/loklak/api/search/LocationWiseTimeServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.json.JSONObject;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class LocationWiseTimeServiceTest {
+
+    @Test
+    public void apiPathTest() {
+
```
[the remark this answers is not in the document]
Comment (simsausaurabh) [reply test/org/loklak/api/search/LocationWiseTimeServiceTest.java]: Working on it.

File: test/org/loklak/api/search/LocationWiseTimeServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @Test
+    public void locationWiseTimeTest() {
+        LocationWiseTimeService locationWiseTimeService = new LocationWiseTimeService();
+        SusiThought susiThought = locationWiseTimeService.locationWiseTime("Chennai");
+        JSONArray jsonArray = susiThought.getData();
+
+        for(int i=0; i<jsonArray.length(); i++) {
+            JSONObject obj = new JSONObject(jsonArray.get(i).toString());
+            assertNotNull(obj.getString("location"));
+            assertNotNull(obj.getString("time"));
```
[the remark this answers is not in the document]
Comment (simsausaurabh) [reply test/org/loklak/api/search/LocationWiseTimeServiceTest.java]: @singhpratyush only two parameters are being received for testing(location and time) and will it be favourable to use any third site/application for checking additional timezone of places? The given two assertions are applicable for all places present in jsonArray.

**Comment by simsausaurabh** [comment]
@singhpratyush please review. I have not included more test parameters because only 2 parameters are being fetched - "location" and "time" from LocationWiseTimeService.java file. And for asserting more tests we would require external 3rd party api/application which would not be favourable.

**Comment by singhpratyush** [comment]
But the time difference between two places (say Chennai and California City) will always be the same. We can use this test to ensure that the parsers are working fine.

Or we can just check for some regex to ensure that proper format is fetched.

In the best case, we can include two more cases for this purpose.

**Comment by simsausaurabh** [comment]
@singhpratyush @jigyasa-grover @daminisatya @sudheesh001 Here I am asserting both pattern checking of time being fetched and time difference of location. Although the test case is passing locally for the pattern check of time, but the travis fails for the pattern match.

Here is the result: 
```
org.loklak.api.search.LocationWiseTimeServiceTest > locationWiseTimeTest FAILED
    java.lang.AssertionError at LocationWiseTimeServiceTest.java:55
```

**Comment by singhpratyush** [comment]
@simsausaurabh: I think a little debugging can get you through it.

Also instead of adding more and more tests in the same function, you should add more test functions.

**Comment by simsausaurabh** [comment]
Thanks @singhpratyush I am looking into it. Yes, I will add more test functions.

**Comment by simsausaurabh** [comment]
@singhpratyush please review.

**Comment by singhpratyush** [comment]
As said earlier,

>  instead of adding more and more tests in the same function, you should add more test functions.

Please update.

**Comment by simsausaurabh** [comment]
@singhpratyush please review.

**Pull request merged**